### PR TITLE
Per-E2E test VM cleanup

### PR DESCRIPTION
**Pull request by maxdrib**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/2320

*Description of changes:*
This PR cleans up VM's at the individual test level to

1. make them more idempotent
2. fix Cloudstack E2E test cleanup, since there's currently no network connection between the Cloudstack environment and the EKS-A build AWS account

In order to avoid creating a cyclic dependency, it required refactoring the cleanup functions into a new package called "cleanup"

*Testing (if applicable):*
Ran TestCloudStackKubernetes121SimpleFlow test case locally with T_CLEANUP_VMS env var set and observed cleanup invocation even though there was nothing to clean up with return status 0

```
2022-06-06T16:24:26.127-0400	V4	Tasks completed	{"duration": "9m31.588544976s"}
2022-06-06T16:24:26.161-0400	V3	Cleaning up long running container	{"name": "eksa_1654546492709757000"}
2022-06-06T16:24:26.847-0400	V2	e2e	Pulling docker image	{"image": "public.ecr.aws/l0g8r8j6/eks-anywhere-cli-tools:v0.7.2-eks-a-v0.0.0-dev-build.1864"}
2022-06-06T16:24:27.940-0400	V3	e2e	Initializing long running container	{"name": "eksa_1654547066847125000", "image": "public.ecr.aws/l0g8r8j6/eks-anywhere-cli-tools:v0.7.2-eks-a-v0.0.0-dev-build.1864"}
2022-06-06T16:24:31.436-0400	V0	e2e	✅ Connected to CloudStack server
2022-06-06T16:24:32.097-0400	V3	e2e	Cleaning up long running container	{"name": "eksa_1654547066847125000"}
    cluster.go:456: failed to clean up VMs: virtual machines for cluster eksa-test-a2c976d not found
2022-06-06T16:24:32.469-0400	V3	e2e	Cleaning up long running container	{"name": "eksa_1654545762898905000"}
--- PASS: TestCloudStackKubernetes121SimpleFlow (1309.85s)
PASS
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/simpleflow_test.go [hunk 1]
```diff
@@ -185,6 +185,7 @@ func TestCloudStackKubernetes121SimpleFlow(t *testing.T) {
 		framework.NewCloudStack(t, framework.WithCloudStackRedhat121()),
 		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube121)),
 	)
+	defer test.CleanupVms()
```
[the remark this answers is not in the document]
Comment (maxdrib) [reply test/e2e/simpleflow_test.go:188]: Good point

File: test/framework/cluster.go [hunk 2]
```diff
@@ -433,6 +437,26 @@ func (e *ClusterE2ETest) DeleteCluster(opts ...CommandOpt) {
...
 }
 
+func (e *ClusterE2ETest) CleanupVms() {
+	shouldCleanupVms, err := getCleanupVmsVar()
+	if err != nil || !shouldCleanupVms {
+		e.T.Logf("Skipping VM cleanup")
+		return
+	}
+
+	switch e.Provider.Name() {
```
[the remark this answers is not in the document]
Comment (maxdrib) [reply test/framework/cluster.go:447]: Excellent suggestion

File: test/framework/cluster.go [hunk 3]
```diff
@@ -433,6 +440,19 @@ func (e *ClusterE2ETest) DeleteCluster(opts ...CommandOpt) {
...
 
+func (e *ClusterE2ETest) CleanupVms() {
+	shouldCleanupVms, err := getCleanupVmsVar()
+	if err != nil || !shouldCleanupVms {
+		e.T.Logf("Skipping VM cleanup")
+		return
+	}
+
+	err = e.Provider.CleanupVMs(e.ClusterName)
+	if err != nil {
```
Comment (g-gaston) [inline test/framework/cluster.go:451]:
```suggestion
	if err = e.Provider.CleanupVMs(e.ClusterName); err != nil {
```

File: test/framework/cluster.go [hunk 4]
```diff
@@ -433,6 +440,19 @@ func (e *ClusterE2ETest) DeleteCluster(opts ...CommandOpt) {
 	e.deleteCluster(opts...)
 }
 
+func (e *ClusterE2ETest) CleanupVms() {
+	shouldCleanupVms, err := getCleanupVmsVar()
+	if err != nil || !shouldCleanupVms {
```
Comment (g-gaston) [inline test/framework/cluster.go:445]: super nit: I would maybe abstract this in a separate function like

```go
func shouldCleanUpVms() bool {
    shouldCleanupVms, err := getCleanupVmsVar()
    return err == nil && shouldCleanupVms
}
```

**Review by g-gaston — APPROVED** [review]

File: test/framework/cluster.go [hunk 5]
```diff
@@ -93,6 +95,10 @@ func NewClusterE2ETest(t *testing.T, provider Provider, opts ...ClusterE2ETestOp
 		opt(e)
 	}
 
+	e.T.Cleanup(func() {
```
Comment (vivek-koppuru) [inline test/framework/cluster.go:98]: We should set this cleanup function once we know that everything on the provider is set up to create the VMs right? That would mean moving it a line below.

**Review by vivek-koppuru — APPROVED** [review]

File: test/framework/cluster.go (hunk 5 above)
Comment (maxdrib) [reply test/framework/cluster.go:100]: done thanks!

**Review by vivek-koppuru — APPROVED** [review]

**Review by vivek-koppuru — APPROVED** [review]
/approve

**Pull request merged**